### PR TITLE
Feature/test room

### PR DESCRIPTION
**Pull request by souziel**

### CODE REVIEW

**Review by guyts96 — APPROVED** [review]

File: server/src/controllers/roomController.ts [hunk 1]
```diff
@@ -7,8 +7,12 @@ class RoomController {
 
     @App.get('')
     async getRooms(req: Request, res: Response): Promise<void> {
-        const result = await roomService.getRooms();
-        res.status(200).send(result);
+        try {
```
Comment (Lukyio) [inline server/src/controllers/roomController.ts:10]: 👍

File: server/src/controllers/roomController.ts [hunk 2]
```diff
@@ -17,19 +21,25 @@ class RoomController {
...
     async getRoom(req: Request, res: Response): Promise<void> {
         const areaId = req.params.roomId;
-        const result = await roomService.getRoomById(areaId).catch(() => undefined);
-        if (result === undefined) {
-            res.status(400).send({ message: 'Error while getting room' });
-            return;
+        try {
+            const result = await roomService.getRoomById(areaId);
+            const code = result ? 200 : 404;
+            const data = result ?? { message: 'No room with such id' };
```
Comment (Lukyio) [inline server/src/controllers/roomController.ts:38]: What does "??" means ?

File: server/test/roomController.test.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    roomId: expect.any(String),
+    name: expect.any(String),
+    devices: expect.any(Array),
+    automations: expect.any(Array),
+} as Room);
+
+const baseUrl: string = 'http://localhost:3000';
+
+describe('Create room controller test', () => {
+    test('Should can\'t create room with no name', async () => {
```
Comment (Lukyio) [inline server/test/roomController.test.ts]: For the test name, you can rather put: 

Should not create ...

File: server/src/controllers/roomController.ts (hunk 2 above)
Comment (guyts96) [reply server/src/controllers/roomController.ts:38]: If result is undefined or null, then it will be replaced with what follows '??'.

**Review by Lukyio — APPROVED** [review]
♌

**Review by Plantin7 — APPROVED** [review]

**Pull request merged**